C++ 20 Compatible

If compile with `-std=c++20`

got an error:

```
In file included from request.cpp:1:
In file included from cmake-build-debug/_deps/ozo-src/include/ozo/connection_info.h:5:
In file included from cmake-build-debug/_deps/ozo-src/include/ozo/impl/async_connect.h:6:
In file included from cmake-build-debug/_deps/ozo-src/include/ozo/impl/io.h:6:
cmake-build-debug/_deps/ozo-src/include/ozo/io/binary_query.h:131:66: error: no member named 'rebind' in 'std::__1::allocator<void>'
                                    typename Allocator::template rebind<char>::other>;
                                             ~~~~~~~~~~~         ^
```

because of `rebind` and other helpers was removed from library (`(deprecated in C++17)(removed in C++20)`):

http://www.open-std.org/Jtc1/sc22/wg21/docs/papers/2018/p0619r3.html
https://en.cppreference.com/w/cpp/memory/allocator

--- a/include/ozo/io/binary_query.h
+++ b/include/ozo/io/binary_query.h
@@ -11,6 +11,8 @@
 #include <ozo/optional.h>
 #include <ozo/pg/types.h>
 
+#include <boost/core/allocator_access.hpp>
+
 #include <boost/hana/for_each.hpp>
 #include <boost/hana/tuple.hpp>
 #include <boost/hana/ext/std/array.hpp>
@@ -128,7 +130,7 @@ private:
         using allocator_type = std::conditional_t<
                                 std::is_same_v<typename Allocator::value_type, char>,
                                     Allocator,
-                                    typename Allocator::template rebind<char>::other>;
+                                    boost::allocator_rebind_t<Allocator, char>>;
         using buffer_type = std::vector<char, allocator_type>;
         using oid_map_type = OidMap;
         using text_type = std::decay_t<Text>;

--- a/include/ozo/io/binary_query.h
+++ b/include/ozo/io/binary_query.h
@@ -11,8 +11,6 @@
 #include <ozo/optional.h>
 #include <ozo/pg/types.h>
 
-#include <boost/core/allocator_access.hpp>
-
 #include <boost/hana/for_each.hpp>
 #include <boost/hana/tuple.hpp>
 #include <boost/hana/ext/std/array.hpp>
@@ -130,7 +128,7 @@ private:
         using allocator_type = std::conditional_t<
                                 std::is_same_v<typename Allocator::value_type, char>,
                                     Allocator,
-                                    boost::allocator_rebind_t<Allocator, char>>;
+                                    typename std::allocator_traits<Allocator>::template rebind_alloc<char>>;
         using buffer_type = std::vector<char, allocator_type>;
         using oid_map_type = OidMap;
         using text_type = std::decay_t<Text>;

